Util/BeatmapsUtil.cs

```diff
@@ -1,19 +1,28 @@
 ﻿namespace BetterSongList.Util {
 	static class BeatmapsUtil {
-		public static string GetHashOfPreview(IPreviewBeatmapLevel preview) {
-			if(preview.levelID.Length < 53)
+
+		public static string GetHashOfLevel(BeatmapLevel level) {
+			return level == null ? null : GetHashOfLevelId(level.levelID);
+		}
+		
+		public static string GetHashOfBeatmapKey(BeatmapKey key) {
+			return GetHashOfLevelId(key.levelId);
+		}
+		
+		private static string GetHashOfLevelId(string id) {
+			if(id.Length < 53)
 				return null;
 
-			if(preview.levelID[12] != '_') // custom_level_<hash, 40 chars>
+			if(id[12] != '_') // custom_level_<hash, 40 chars>
 				return null;
 
-			return preview.levelID.Substring(13, 40);
+			return id.Substring(13, 40);
 		}
 
-		public static int GetCharacteristicFromDifficulty(IDifficultyBeatmap diff) {
-			var d = diff.parentDifficultyBeatmapSet?.beatmapCharacteristic.sortingOrder;
+		public static int GetCharacteristicFromDifficulty(BeatmapKey diff) {
+			var d = diff.beatmapCharacteristic.sortingOrder;
 
-			if(d == null || d > 4)
```

Comment:
Why was this null check removed?

Reply:
Because `BeatmapKey` is a struct, and not nullable unless using `BeatmapKey?`

Reply:
Oh wait you are right (well kind of), its an int. I must've overlooked that (or they changed it). Ok 👍

Reply:
Actually no nevermind, the characteristic could still be a null there. I'll just add it back in to be safe

Reply:
oh yeah it is an int. pepega me was talking about a different variable.